src/metabase/query_processor/middleware/annotate.clj

```diff
@@ -143,12 +151,12 @@
          :display_name (humanization/name->human-readable-name field-name)})
 
     [:expression expression-name]
-    {:name            expression-name
-     :display_name    expression-name
-     :base_type       :type/Float
-     :special_type    :type/Number
-     ;; provided so the FE can add easily add sorts and the like when someone clicks a column header
-     :expression_name expression-name}
+    (merge
+     (infer-expression-type (-> expression-name keyword expressions))
```

Comment:
is there any way that this could be made to work if expression names as either strings or keywords?

It really doesn't seem like it makes sense to have expression names be keywords, especially since they can be things like `"My Coool   ex/pre/ssion"` which can't be typed out as a keyword literal.

I've been meaning to fix it for a while, I've had a TODO note in the schema and just opened https://github.com/metabase/mbql/issues/5 to track it. So once that gets fixed it would be nice if it didn't break too many things in Metabase itself